samples/javascript_nodejs/21.corebot-app-insights/dialogs/mainDialog.js

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ComponentDialog, DialogSet, DialogTurnStatus, TextPrompt, WaterfallDialog } = require('botbuilder-dialogs');
+const { LuisRecognizer } = require('botbuilder-ai');
+const { TimexProperty } = require('@microsoft/recognizers-text-data-types-timex-expression');
+const { MessageFactory, InputHints } = require('botbuilder');
+
+const MAIN_WATERFALL_DIALOG = 'mainWaterfallDialog';
+
+class MainDialog extends ComponentDialog {
+    /**
+     * @param {*} luisRecognizer
+     * @param {*} bookingDialog
+     */
+    constructor(luisRecognizer, bookingDialog) {
+        super('MainDialog');
+
+        if (!luisRecognizer) { throw new Error('[MainDialog]: Missing parameter \'luisRecognizer\' is required.'); }
+        if (!bookingDialog) { throw new Error('[MainDialog]: Missing parameter \'bookingDialog\' is required.'); }
+
+        this.luisRecognizer = luisRecognizer;
+        this.bookingDialog = bookingDialog;
```

Comment:
bookingDialog is only used in the constructor, so it does not need to be set here.

```suggestion
```